### PR TITLE
Added Tofts and Ext. Tofts

### PR DESCRIPTION
**Pull request by stadmill**
Added functionality for Tofts and ETofts modelling plus tests, documentation and examples. Merge with main will update the website to show changes.

### CODE REVIEW

File: src/osipi/_tissue.py
Comment (ltorres6) [inline src/osipi/_tissue.py:1]: Nice documentation. I would swap the warnings package out with the logging package, which will also replace any print statements with better logging methods for INFO, DEBUG, WARNING, and ERROR logging. I can help with properly setting this up across the package if wanted.

**Review by ltorres6 — COMMENTED** [review]
Okay I've gone through the MR and have some notes/suggestions.

Overall Structure
- Nice job on setting up Sphinx and automated publishing. The docs look nice.
- I would change the name of the repository to  "osipi" or "osipi_sdk" rather than pypi. 
- I would change the name of plot_dummy to plot_template or something similar, indicating it's meant to provide a starting point for something rather than a dummy placeholder file.

It looks like the package structure is designed to align with the lexicon, which is great wrt nomenclature. Maybe it would be a good idea to include references to the relevant lexicon pages when submitting a pull request so that a reviewer can use that as a reference (just thinking out loud here). 

Probably out of scope, but I'm not sure the package structure itself makes sense from a users POV. If I work in DCE, I don't want to sift through all of the functions in the package to find relevant models, discretization methods, etc. I think the following structure for submodules makes more sense: osipi.dce, osipi.asl, osipi.dsc could all be entrypoints for a specific set of tools in that field. osipi.utils can contain anything worth generalizing, like generic deconvolution/convolution methods or all of the signal models (GRE, SE, SPGR, etc). I think this would increase modularity, reusability, and user understanding. For example, the DRO will be able to leverage all of the signal models to generate specific contrasts. If it were me, I'd create an osipi.dro sub-module for this development, for example. 

Anyway, nice job. Feel free to tag me on future PR's and shoot me a message and I'll take a look.

File: docs/.DS_Store
[the remark this answers is not in the document]
Comment (stadmill) [reply docs/.DS_Store:1]: I am confused. Should I add the files to ignore in the .gitignore I have locally? Or is this a setting for the repo on Github?

File: README.md
[the remark this answers is not in the document]
Comment (stadmill) [reply README.md:1]: I think this is a good idea. I will add this as an issue and make it independent of this PR.

File: src/osipi/_tissue.py
Comment (stadmill) [reply src/osipi/_tissue.py:1]: As above. I will add this as an issue. Thanks for offering to set this up. I will assign it to you.

File: docs/.DS_Store
[the remark this answers is not in the document]
Comment (ltorres6) [reply docs/.DS_Store:1]: You can add a .gitignore to the repository itself with these files ignored.

File: .gitignore
Comment (ltorres6) [inline .gitignore:1]: You can keep this file on the repo. You can add .DS_Store as well.

**Review by ltorres6 — APPROVED** [review]
Left one comment, but otherwise looks good to me with all of the issues you've created using the issue tracker.

**Comment by stadmill** [comment]
Something wrong with Sphinx or make. I think there is some error during 'make html'. It says it can't find osipi...I am able to build the html without errors locally, so not sure why it's popping up on Github. Do you think this has to do with me deleting all the DS_Store files on the repo?!

**Comment by ltorres6** [comment]
The issue is with osipi not being able to load due to scipy being a dependency, and not being installed. I was able to replicate the error locally in a clean environment. I will make an MR for the fix to this and add you as a reviewer.

**Pull request merged**